sql-common/client.c

```diff
@@ -802,7 +802,7 @@ void read_ok_ex(MYSQL *mysql, ulong length) {
             if (is_charset == 1) {
               char charset_name[MY_CS_NAME_SIZE * 8]; // MY_CS_BUFFER_SIZE
               size_t length = data->length > (sizeof(charset_name) - 1)
-                                  ? sizeof(charset_name - 1)
+                                  ? (sizeof(charset_name) - 1)
```

Comment:
This looks like an upstream bug - should be reported separately.
Please, create a PS-NNN bug, an upstream bug and specify in the commit message `this merge commit also fixes PS-NNN`.

Reply:
Reported at:
https://bugs.mysql.com/bug.php?id=96433
https://jira.percona.com/browse/PS-5828